language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/FromJsonGenerator.scala

```diff
@@ -227,7 +232,7 @@ private[inner] object FromJsonGenerator extends StrictLogging {
 
     damlType match {
       case TypeCon(TypeConName(ident), typeParams) =>
-        CodeBlock.of("$T.jsonDecoder($L)", guessClass(ident), typeReaders(typeParams))
+        CodeBlock.of("$L.jsonDecoder($L)", guessClass(ident).simpleName, typeReaders(typeParams))
```

Comment:
The type of `guessClass(ident)` should have already been imported as a part of defining the corresponding field, so we  refer to the class by `simpleName`, to avoid "obscuring" of the package name with the field name.
https://docs.oracle.com/javase/specs/jls/se7/html/jls-6.html#jls-6.4.2